tests/test_base.py

```diff
@@ -69,6 +69,169 @@
+    np.testing.assert_array_less(1e100,delta.stepK())
+    # Check input flux vs output flux
+    for inFlux in np.logspace(-2, 2, 10):
+        delta = galsim.DeltaFunction(flux=inFlux)
+        outFlux = delta.getFlux()
+        np.testing.assert_almost_equal(outFlux, inFlux)
+
+@timer
+def test_deltaFunction_flux_scaling():
+    """Test flux scaling for Delta function.
+    """
+    # decimal point to go to for parameter value comparisons
+    param_decimal = 12
+
+    # init with flux only (should be ok given last tests)
+    obj = galsim.DeltaFunction(flux=test_flux)
+    obj *= 2.
+    np.testing.assert_almost_equal(
+        obj.getFlux(), test_flux * 2., decimal=param_decimal,
+        err_msg="Flux param inconsistent after __imul__.")
+    obj = galsim.DeltaFunction(flux=test_flux)
+    obj /= 2.
+    np.testing.assert_almost_equal(
+        obj.getFlux(), test_flux / 2., decimal=param_decimal,
+        err_msg="Flux param inconsistent after __idiv__.")
+    obj = galsim.DeltaFunction(flux=test_flux)
+    obj2 = obj * 2.
+    # First test that original obj is unharmed...
+    np.testing.assert_almost_equal(
+        obj.getFlux(), test_flux, decimal=param_decimal,
+        err_msg="Flux param inconsistent after __rmul__ (original).")
+    # Then test new obj2 flux
+    np.testing.assert_almost_equal(
+        obj2.getFlux(), test_flux * 2., decimal=param_decimal,
+        err_msg="Flux param inconsistent after __rmul__ (result).")
+    obj = galsim.DeltaFunction(flux=test_flux)
+    obj2 = 2. * obj
+    # First test that original obj is unharmed...
+    np.testing.assert_almost_equal(
+        obj.getFlux(), test_flux, decimal=param_decimal,
+        err_msg="Flux param inconsistent after __mul__ (original).")
+    # Then test new obj2 flux
+    np.testing.assert_almost_equal(
+        obj2.getFlux(), test_flux * 2., decimal=param_decimal,
+        err_msg="Flux param inconsistent after __mul__ (result).")
+    obj = galsim.DeltaFunction(flux=test_flux)
+    obj2 = obj / 2.
+    # First test that original obj is unharmed...
+    np.testing.assert_almost_equal(
+        obj.getFlux(), test_flux, decimal=param_decimal,
+        err_msg="Flux param inconsistent after __div__ (original).")
+    # Then test new obj2 flux
+    np.testing.assert_almost_equal(
+        obj2.getFlux(), test_flux / 2., decimal=param_decimal,
+        err_msg="Flux param inconsistent after __div__ (result).")
+        
+    obj = galsim.DeltaFunction(flux=test_flux)
+    obj2 = obj.withFlux(test_flux*2.)
+    # First test that original obj is unharmed...
+    np.testing.assert_almost_equal(
+        obj.getFlux(), test_flux, decimal=param_decimal,
+        err_msg="Flux param inconsistent after obj.withFlux(flux).")
+    # Then test new obj2 flux
+    np.testing.assert_almost_equal(
+        obj2.getFlux(), test_flux * 2., decimal=param_decimal,
+        err_msg="Flux param inconsistent after obj.withFlux(flux).")
+        
```

Comment:
Some other things that might be worth testing:
* I don't see any tests of the shooting code.  Should add a call to `do_shoot` for a DeltaFunction convolved with something simple.  
* Maybe convolve more than one DeltaFunction together.  fluxes should multiply.
* Rather than test that expand, shear, etc. give errors, I think they should be no ops, so check that a sheared DeltaFunction convolved with something is equivalent to not shearing it.  (shearing just the DeltaFunction that is, not the other component.)
* There is a function called `check_basic` that does a number of sanity checks.  It's possible that DeltaFunction may fail for this, since it might make assumptions about the way objects work that don't apply here.  But at least try it out and if necessary add a kwarg to skip the basic tests that don't really make sense for DeltaFunction.
* Maybe try a real-space convolution, especially if you try the thing I suggested above about making xValue near zero not inf, but rather flux / eps^2.  You can test this with `Convolve(delta, gauss, real_space=True)`.
* I don't think you anywhere construct `DeltaFunction()` with no arguments.  Should do that and make sure it has flux=1.  Also make sure it pickles correctly, which will cover that branch in `__str__`.